internal/credentials/ecr/pod_identity.go

```diff
@@ -0,0 +1,146 @@
+package ecr
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"os"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/patrickmn/go-cache"
+
+	"github.com/akuity/kargo/internal/logging"
+)
+
+var ecrURLRegex = regexp.MustCompile(`^([0-9]{12})\.dkr\.ecr\.(.+)\.amazonaws\.com`)
```

Comment:
Is this true for government and AWS (Top) Secret cloud environments?

Reply:
Good question. After a bit of searching, I am no closer to any clarity on this. I think if this doesn't hold true for those scenarios, we'll hear about it soon enough.